packages/ruleset/src/functions/operation-id-case-convention.js

```diff
@@ -0,0 +1,35 @@
+const { casing } = require('@stoplight/spectral-functions');
+const { isSdkExcluded } = require('../utils');
+let casingConfig;
+
+module.exports = function(operation, options, { path }) {
+  // Save this rule's "functionOptions" value since we need
+  // to pass it on to Spectral's "casing" function.
+  casingConfig = options;
+
+  return operationIdCaseConvention(operation, path);
+};
+
+function operationIdCaseConvention(operation, path) {
+  // Bypass the check if the operation is excluded or if the
+  // operationId value is missing (the existence of an operationId value
+  // is checked by the 'operation-operationId' rule).
+  const operationId =
+    operation.operationId && operation.operationId.toString().trim();
+  if (isSdkExcluded(operation) || !operationId) {
+    return [];
+  }
+
+  const errors = [];
+
+  const result = casing(operationId, casingConfig);
+  if (result) {
+    // Update the message to clarify that it is an operationId value that is incorrect,
+    // and update the path.
+    result[0].message = 'Operation ids ' + result[0].message;
+    result[0].path = [...path, 'operationId'];
+    errors.push(result[0]);
```

Comment:
Not important but I thought I had for these operation rules that don't loop through anything: we could return a single value when the rule is broken and avoid defining the array, like
```suggestion
    return [result[0]];
```

and at the end of the function, `return [];`.

I really don't know how much of a performance implication there actually is for allocating the array every time so feel free to ignore this, just a suggestion.

Reply:
fixed that